engines/prover.cpp

```diff
@@ -68,10 +70,284 @@ void Prover::initialize()
 {
   reached_k_ = -1;
   bad_ = solver_->make_term(smt::PrimOp::Not, property_.prop());
+  if (options_.static_coi_) {
+    if (!ts_.is_functional())
+      throw PonoException("Temporary restriction: cone-of-influence analysis currently "\
+                          "supported for functional transition systems only.");
+    /* Compute the set of state/input variables related to the
+       bad-state property. Based on that information, rebuild the
+       transition relation of the transition system. */
+    compute_coi();
+    orig_num_statevars_ = ts_.statevars().size();
+    orig_num_inputvars_ = ts_.inputvars().size();
+    ts_.rebuild_trans_based_on_coi(statevars_in_coi_, inputvars_in_coi_);
+    assert(statevars_in_coi_.size() == ts_.statevars().size());
+    assert(inputvars_in_coi_.size() == ts_.inputvars().size());
+    logger.log(
+        1,
+        "COI analysis completed: {} remaining input variables, {} original",
+        inputvars_in_coi_.size(),
+        orig_num_inputvars_);
+    logger.log(
+        1,
+        "COI analysis completed: {} remaining state variables, {} original",
+        statevars_in_coi_.size(),
+        orig_num_statevars_);
+  }
 }
 
 ProverResult Prover::prove() { return check_until(INT_MAX); }
 
+/* For debugging only. */
+void Prover::print_term_dfs(const Term & term)
```

Comment:
this could be a static non-member function

Reply:
I agree. I suggest to leave it for now as it will change during refactoring.